### DISCUSSION TITLE
Optimizing for comfort and resource usage using ems-esp32 and a home automation software

### DISCUSSION THREAD

**Discussion by kongomongo** [emsesp/EMS-ESP32#2213] [Q&A] [answered]
Hello everyone,

thanks for this awesome project. I have got it up and running in my openhab setup and made a small grafana dashboard of some of the values that seem relevant to me (hard to pick with > 120 sensor entries ;-)

OK so I have a Buderus Logamax Buderus GB172-14 with a RC310 as well as a TWL KER 500 storage tank with underfloor heating. 

So everything works OK so far with one exception. Everytime multiple people shower or someone takes a bath dhw temperature first drops to a level that does not feel warm anymore after like 2 minutes. 

After 10 minutes some logic seems to kick in and lets the burner run at 100% which raises the temperatures to comfortable levels for a shower/bath again.

Now I am in a bit of a dilemma: I don't really want to raise the general temperature level, I think 45-50° is fine for warm water, but I do not want my showers/baths to be "luke warm". 

So I see 2 options:
- Optimize my config so it just always works -> I need suggestion from experienced people what's wrong in my setup
- I lower the temperature/resource usage even further and need to use a "charge" button from my home automation that sends commands to the ems-esp32 to guarantee warm temps for 5-10 minutes.

Here is some data and some pictures for general insight:

[Boiler_GBx72_Trendline_Cerapur_Greenstar Si.csv](https://github.com/user-attachments/files/17789618/Boiler_GBx72_Trendline_Cerapur_Greenstar.Si.csv)
[Thermostat_RC300_RC310_Moduline 3000_1010H_CW400_Sense II_HPC410.csv](https://github.com/user-attachments/files/17789619/Thermostat_RC300_RC310_Moduline.3000_1010H_CW400_Sense.II_HPC410.csv)
[Mixer Module_MM100.csv](https://github.com/user-attachments/files/17789620/Mixer.Module_MM100.csv)

![image](https://github.com/user-attachments/assets/fbda3e14-9967-439d-b7f3-04364f920d52)

the red bar indicates start of taking a bath
![image](https://github.com/user-attachments/assets/66db9519-d5cd-4e23-8b74-e4157f2af6b3)

**Comment by kongomongo** [comment]
Vorlauf = curflowtemp
Vorlauf HK = flowtemphc
Außen = outdoortemp
Mischer = switchtemp
WW intern = wwcurtemp
WW extern = wwcurtemp2
WW Speicher = wwstoragetemp2
Heizung aktiv = heatingactive
WW aktiv = tapwateractive
Energie Heizung = nrgheat
Energie WW = nrgww
Brennerleistung = curburnpow

**Reply by MichaelDvP** [reply, marked as answer]
You have set `dhw comfort;eco`, 
Set it to hot and the logic will always recharge if the dhw temperature goes below 
dhw settemp + dhw hysteresis on in your setting settemp 55, hysteresis -5, so dhw is always between 50 and 55.

**Reply by kongomongo** [reply]
oh wow, thank you so much!

must have been set by my installer. is there an explanation for the choices? (eco, hot, intelligent) ?

**Reply by kongomongo** [reply]
Also another question, if I set my settemp to even lower but have a triggered (by home automation or by timer) action that temporarily adjusts for a higher shower/bath temp would that be sensible energy usage wise? Which settings would I adjust in the home automation action? just dhw comfort and settemp?

Thanks again.

**Reply by MichaelDvP** [reply]
Don't know, just try what settings fits best to your needs. You have a thermostat, this will mainly control the settings (and may overwrite your settings on the boiler). You have to send commands to the thermostat. There you can also trigger extra charges.

**Reply by kongomongo** [reply]
OK thank you very much ;-). Sorry to bother you so much but allow me one final question for my understanding:

There are 3 things happening here I don't quite understand:

A) boiler is set to 22°C target floor heating temp till 23:00 then allow 18°C till 5:00. That is probably what makes the burner fixed at 23% power and also allows the burner to shutdown 3 times at this night. Why that percentage and why those times I am unsure.

B) Followed your suggestion and set dhw comfort to "hot" and you can see it switches from "heating active on" to "heating active off" and from "tapwater active off" to "tapwater active on" (going from red background to blue in grafana). So I guess this seems to work.

C) here I set the "heating temperature" [heatingtemp] from 50°C to 45°C because I thought my floor heating only needs ~35°C temp so why have it higher anyways? Unfortunately this also decreased the dhw/tapwater to 45°C as well although "dhw set temperature " is 50°C in boiler as well as theromostat. Is this because of set temp is 50°C and dhw hysteresis is -5°C ? 
If that is the case, would lowering the heatingtemp from 45°C to 40°C make the boiler switch from "heating active" to "tapwater active" as it now needs the heat for the tapwater heating? That would be my expectation.

![image](https://github.com/user-attachments/assets/9c8f85d2-645a-47ef-978c-e78da97f798a)

**Reply by MichaelDvP** [reply]
The boiler models differ sometimes, not sure how your system works. 
Normally if you like to change setpoint values you have to tell it to the thermostat. For heating the thermostat calculates the needed flowtemp and tell the mixer. The mixer corrects this by an offset and tellt to boiler. The boiler controlls the burner to supply this flowtemp.
You can not change the flowtemp on the boiler, it will be overwritten. But you can set parameters like hysteresis, power limit, etc in the boiler. Or the flowtempoffset in the mixer.
heatingtemp is mostly a limit for the boiler temperature, normally set to 90.
